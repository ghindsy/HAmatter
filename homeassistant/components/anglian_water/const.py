"""Anglian Water consts."""

from logging import Logger, getLogger

LOGGER: Logger = getLogger(__package__)

NAME = "Anglian Water"
DOMAIN = "anglian_water"

CONF_DEVICE_ID = "device_id"
