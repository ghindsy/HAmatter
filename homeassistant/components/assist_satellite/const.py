"""Constants for assist satellite."""

DOMAIN = "assist_satellite"
