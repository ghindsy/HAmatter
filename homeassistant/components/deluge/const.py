"""Constants for the Deluge integration."""

import enum
import logging
from typing import Final

CONF_WEB_PORT = "web_port"
CURRENT_STATUS = "current_status"
DEFAULT_NAME = "Deluge"
DEFAULT_RPC_PORT = 58846
DEFAULT_WEB_PORT = 8112
DOMAIN: Final = "deluge"
DOWNLOAD_SPEED = "download_speed"

LOGGER = logging.getLogger(__package__)
PROTOCOL_TRAFFIC_DOWNLOAD_SPEED = "protocol_traffic_download_speed"
PROTOCOL_TRAFFIC_UPLOAD_SPEED = "protocol_traffic_upload_speed"
UPLOAD_SPEED = "upload_speed"


class DelugeGetSessionStatusKeys(enum.Enum):
    """Enum representing the keys that get passed into the Deluge RPC `core.get_session_status` xml rpc method.

    you can call `core.get_session_status` with no keys (so an empty list in deluge-client.DelugeRPCClient.call)
    to get the full list of possible keys, but it seems to basically be a all of the session statistics
    listed on this page: https://www.rasterbar.com/products/libtorrent/manual-ref.html#session-statistics
    and a few others

    there is also a list of deprecated keys that deluge will translate for you and issue a warning in the log:
    https://github.com/deluge-torrent/deluge/blob/7f3f7f69ee78610e95bea07d99f699e9310c4e08/deluge/core/core.py#L58

    """

    DHT_DOWNLOAD_RATE = "dht_download_rate"
    DHT_UPLOAD_RATE = "dht_upload_rate"
    DOWNLOAD_RATE = "download_rate"
    UPLOAD_RATE = "upload_rate"
