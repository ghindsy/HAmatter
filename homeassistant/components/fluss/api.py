"""Fluss+ API Client."""

from __future__ import annotations

import asyncio
import datetime
import logging
import socket
import typing

import aiohttp

from homeassistant.core import HomeAssistant
from homeassistant.helpers import aiohttp_client

LOGGER = logging.getLogger(__package__)


class FlussApiClientError(Exception):
    """Exception to indicate a general API error."""


class FlussDeviceError(Exception):
    """Exception to indicate that an error occurred when retriveing devices."""


class FlussApiClientCommunicationError(FlussApiClientError):
    """Exception to indicate a communication error."""


class FlussApiClientAuthenticationError(FlussApiClientError):
    """Exception to indicate an authentication error."""


class FlussApiClient:
    """Sample API Client."""

    def __init__(self, api_key: str, hass: HomeAssistant) -> None:
        """Sample API Client."""
        self._api_key = api_key
        self._session = aiohttp_client.async_get_clientsession(hass)

    async def async_get_devices(self) -> typing.Any:
        """Get data from the API."""
        return await self._api_wrapper(
            method="get",
            url="https://zgekzokxrl.execute-api.eu-west-1.amazonaws.com/v1/api/device/list",
            headers={"Authorization": self._api_key},
        )

    async def async_trigger_device(self, deviceId: str) -> typing.Any:
        """Trigger the device."""
        timestamp = int(datetime.datetime.now().timestamp() * 1000)
        return await self._api_wrapper(
            method="post",
            url=f"https://zgekzokxrl.execute-api.eu-west-1.amazonaws.com/v1/api/device/{deviceId}/trigger",
            headers={"Authorization": self._api_key},
            data={"timeStamp": timestamp, "metaData": {}},
        )

    async def _api_wrapper(
        self,
        method: str,
        url: str,
        data: dict | None = None,
        headers: dict | None = None,
    ) -> typing.Any:
        """Get information from the API."""
        try:
            async with asyncio.timeout(10):
                response = await self._session.request(
                    method=method,
                    url=url,
                    headers=headers,
                    json=data,
                )
                if response.status in (401, 403):
                    raise FlussApiClientAuthenticationError(
                        "Invalid credentials",
                    )
                response.raise_for_status()
                return await response.json()

        except TimeoutError as exception:
            raise FlussApiClientCommunicationError(
                "Timeout error fetching information",
            ) from exception
        except (aiohttp.ClientError, socket.gaierror) as exception:
            raise FlussApiClientCommunicationError(
                "Error fetching information",
            ) from exception
        except Exception as exception:  # pylint: disable=broad-except
            raise FlussApiClientError("Something really wrong happened!") from exception
