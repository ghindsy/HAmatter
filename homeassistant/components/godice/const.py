"""Constants for the godice integration."""

from datetime import timedelta

DOMAIN = "godice"
SCAN_INTERVAL = timedelta(seconds=60)

MANUFACTURER = "Particula"
MODEL = "GoDice"

CONF_SHELL = "shell"

DICE_MIN_NUMBER = 1
DICE_MAX_NUMBER = 20
