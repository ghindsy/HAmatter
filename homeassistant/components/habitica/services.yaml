# Describes the format for Habitica service
api_call:
  fields:
    config_entry:
      required: true
      selector:
        config_entry:
          integration: habitica
    path:
      required: true
      example: '["tasks", "user", "post"]'
      selector:
        object:
    args:
      example: '{"text": "Use API from Home Assistant", "type": "todo"}'
      selector:
        object:
