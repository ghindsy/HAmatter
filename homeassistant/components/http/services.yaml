unban_ip:
  fields:
    ip_address:
      required: true
      example: "192.168.178.2"
      selector:
        text:
