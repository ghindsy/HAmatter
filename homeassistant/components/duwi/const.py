"""Constants for the Duwi integration.

This module contains constants that are used throughout the Duwi Smart Hub integration,
including configuration keys, default values, and URLs for API endpoints.
"""

from homeassistant.const import Platform

# Unique domain identifier for the Duwi Smart Hub integration
DOMAIN = "duwi"

# Manufacturer of the product for identification purposes
MANUFACTURER = "Duwi"

# Version of the App for the Duwi Smart Hub integration, used for tracking and compatibility.
APP_VERSION = "0.1.1"

# Home Assistant client version
CLIENT_VERSION = "0.1.1"

# Model identification of the Home Assistant client, typically used for logging or diagnostics.
CLIENT_MODEL = "homeassistant"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [Platform.SWITCH]

SENSOR_TYPE = "7"
