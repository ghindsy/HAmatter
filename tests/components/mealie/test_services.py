"""Tests for the Mealie services."""

from datetime import date
from unittest.mock import AsyncMock

from aiomealie.exceptions import MealieNotFoundError
from freezegun.api import FrozenDateTimeFactory
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.mealie.const import (
    ATTR_CONFIG_ENTRY_ID,
    ATTR_END_DATE,
    ATTR_RECIPE_ID,
    ATTR_START_DATE,
    DOMAIN,
)
from homeassistant.components.mealie.services import (
    SERVICE_GET_MEALPLAN,
    SERVICE_GET_RECIPE,
)
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ServiceValidationError

from . import setup_integration

from tests.common import MockConfigEntry


async def test_service_mealplan(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    freezer: FrozenDateTimeFactory,
) -> None:
    """Test the get_mealplan service."""

    await setup_integration(hass, mock_config_entry)

    freezer.move_to("2023-10-21")

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_MEALPLAN,
        {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id},
        blocking=True,
        return_response=True,
    )
    assert mock_mealie_client.get_mealplans.call_args_list[1][0] == (
        date(2023, 10, 21),
        date(2023, 10, 21),
    )
    assert response == snapshot

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_MEALPLAN,
        {
            ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
            ATTR_START_DATE: date(2023, 10, 22),
            ATTR_END_DATE: date(2023, 10, 25),
        },
        blocking=True,
        return_response=True,
    )
    assert response
    assert mock_mealie_client.get_mealplans.call_args_list[2][0] == (
        date(2023, 10, 22),
        date(2023, 10, 25),
    )

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_MEALPLAN,
        {
            ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
            ATTR_START_DATE: date(2023, 10, 19),
        },
        blocking=True,
        return_response=True,
    )
    assert response
    assert mock_mealie_client.get_mealplans.call_args_list[3][0] == (
        date(2023, 10, 19),
        date(2023, 10, 21),
    )

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_MEALPLAN,
        {
            ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
            ATTR_END_DATE: date(2023, 10, 22),
        },
        blocking=True,
        return_response=True,
    )
    assert response
    assert mock_mealie_client.get_mealplans.call_args_list[4][0] == (
        date(2023, 10, 21),
        date(2023, 10, 22),
    )

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_MEALPLAN,
            {
                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
                ATTR_START_DATE: date(2023, 10, 22),
                ATTR_END_DATE: date(2023, 10, 19),
            },
            blocking=True,
            return_response=True,
        )


async def test_service_recipe(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the get_recipe service."""

    await setup_integration(hass, mock_config_entry)

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_RECIPE,
        {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id, ATTR_RECIPE_ID: "recipe_id"},
        blocking=True,
        return_response=True,
    )
    assert response == snapshot


async def test_service_recipe_not_found(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test the get_recipe service."""

    await setup_integration(hass, mock_config_entry)

    mock_mealie_client.get_recipe.side_effect = MealieNotFoundError

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_RECIPE,
            {
                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
                ATTR_RECIPE_ID: "recipe_id",
            },
            blocking=True,
            return_response=True,
        )


async def test_service_mealplan_without_entry(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test the get_mealplan service without entry."""
    mock_config_entry.add_to_hass(hass)
    mock_config_entry2 = MockConfigEntry(domain=DOMAIN)
    mock_config_entry2.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_MEALPLAN,
            {ATTR_CONFIG_ENTRY_ID: mock_config_entry2.entry_id},
            blocking=True,
            return_response=True,
        )

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_MEALPLAN,
            {ATTR_CONFIG_ENTRY_ID: "bad-config_id"},
            blocking=True,
            return_response=True,
        )
