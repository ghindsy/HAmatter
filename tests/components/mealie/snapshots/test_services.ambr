# serializer version: 1
# name: test_service_mealplan
  dict({
    'mealplan': list([
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 22),
        'mealplan_id': 230,
        'recipe': dict({
          'description': "Een traybake is eigenlijk altijd een goed idee. Deze zoete aardappel curry traybake dus ook. Waarom? Omdat je alleen maar wat groenten - en in dit geval kip - op een bakplaat (traybake dus) legt, hier wat kruiden aan toevoegt en deze in de oven schuift. Ideaal dus als je geen zin hebt om lang in de keuken te staan. Maar gewoon lekker op de bank wil ploffen om te wachten tot de oven klaar is. Joe! That\\'s what we like. Deze zoete aardappel curry traybake bevat behalve zoete aardappel en curry ook kikkererwten, kippendijfilet en bloemkoolroosjes. Je gebruikt yoghurt en limoen als een soort dressing. En je serveert deze heerlijke traybake met naanbrood. Je kunt natuurljk ook voor deze traybake met chipolataworstjes gaan. Wil je graag meer ovengerechten? Dan moet je eigenlijk even kijken naar onze Ovenbijbel. Onmisbaar in je keuken! We willen je deze zoete aardappelstamppot met prei ook niet onthouden. Megalekker bordje comfortfood als je \\'t ons vraagt.",
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'AiIo',
          'name': 'Zoete aardappel curry traybake',
          'original_url': 'https://chickslovefood.com/recept/zoete-aardappel-curry-traybake/',
          'recipe_id': 'c5f00a93-71a2-4e48-900f-d9ad0bb9de93',
          'recipe_yield': '2 servings',
          'slug': 'zoete-aardappel-curry-traybake',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.BREAKFAST: 'breakfast'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 229,
        'recipe': dict({
          'description': 'The BEST Roast Chicken recipe is simple, budget friendly, and gives you a tender, mouth-watering chicken full of flavor! Served with roasted vegetables, this recipe is simple enough for any cook!',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'JeQ2',
          'name': 'Roast Chicken',
          'original_url': 'https://tastesbetterfromscratch.com/roast-chicken/',
          'recipe_id': '5b055066-d57d-4fd0-8dfd-a2c2f07b36f1',
          'recipe_yield': '6 servings',
          'slug': 'roast-chicken',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.LUNCH: 'lunch'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 226,
        'recipe': dict({
          'description': 'Te explicamos paso a paso, de manera sencilla, la elaboración de la receta de pollo al curry con leche de coco en 10 minutos. Ingredientes, tiempo de...',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'INQz',
          'name': 'Receta de pollo al curry en 10 minutos (con vídeo incluido)',
          'original_url': 'https://www.directoalpaladar.com/recetas-de-carnes-y-aves/receta-de-pollo-al-curry-en-10-minutos',
          'recipe_id': 'e360a0cc-18b0-4a84-a91b-8aa59e2451c9',
          'recipe_yield': '2 servings',
          'slug': 'receta-de-pollo-al-curry-en-10-minutos-con-video-incluido',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.LUNCH: 'lunch'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 224,
        'recipe': dict({
          'description': 'bourguignon, oignon, carotte, bouquet garni, vin rouge, beurre, sel, poivre',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'nj5M',
          'name': 'Boeuf bourguignon : la vraie recette (2)',
          'original_url': 'https://www.marmiton.org/recettes/recette_boeuf-bourguignon_18889.aspx',
          'recipe_id': '9c7b8aee-c93c-4b1b-ab48-2625d444743a',
          'recipe_yield': '4 servings',
          'slug': 'boeuf-bourguignon-la-vraie-recette-2',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 222,
        'recipe': dict({
          'description': 'Εύκολη μακαρονάδα με κεφτεδάκια στον φούρνο από τον Άκη Πετρετζίκη. Φτιάξτε την πιο εύκολη μακαρονάδα με κεφτεδάκια σε μόνο ένα σκεύος.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'En9o',
          'name': 'Εύκολη μακαρονάδα με κεφτεδάκια στον φούρνο (1)',
          'original_url': 'https://akispetretzikis.com/recipe/7959/efkolh-makaronada-me-keftedakia-ston-fourno',
          'recipe_id': 'f79f7e9d-4b58-4930-a586-2b127f16ee34',
          'recipe_yield': '6 servings',
          'slug': 'eukole-makaronada-me-kephtedakia-ston-phourno-1',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 221,
        'recipe': dict({
          'description': 'Delicious Greek turkey meatballs with lemon orzo, tender veggies, and a creamy feta yogurt sauce. These healthy baked Greek turkey meatballs are filled with tons of wonderful herbs and make the perfect protein-packed weeknight meal!',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'Kn62',
          'name': 'Greek Turkey Meatballs with Lemon Orzo & Creamy Feta Yogurt Sauce',
          'original_url': 'https://www.ambitiouskitchen.com/greek-turkey-meatballs/',
          'recipe_id': '47595e4c-52bc-441d-b273-3edf4258806d',
          'recipe_yield': '4 servings',
          'slug': 'greek-turkey-meatballs-with-lemon-orzo-creamy-feta-yogurt-sauce',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.SIDE: 'side'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 220,
        'recipe': dict({
          'description': 'Einfacher Nudelauflauf mit Brokkoli, Sahnesauce und extra Käse. Dieses vegetarische 5 Zutaten Rezept ist super schnell gemacht und SO gut!',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'nOPT',
          'name': 'Einfacher Nudelauflauf mit Brokkoli',
          'original_url': 'https://kochkarussell.com/einfacher-nudelauflauf-brokkoli/',
          'recipe_id': '9d553779-607e-471b-acf3-84e6be27b159',
          'recipe_yield': '4 servings',
          'slug': 'einfacher-nudelauflauf-mit-brokkoli',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 219,
        'recipe': dict({
          'description': 'This is a modified Pampered Chef recipe. You can use a trifle bowl or large glass punch/salad bowl to show it off. It is really easy to make and I never have any leftovers. Cook time includes chill time.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'ibL6',
          'name': 'Pampered Chef Double Chocolate Mocha Trifle',
          'original_url': 'https://www.food.com/recipe/pampered-chef-double-chocolate-mocha-trifle-74963',
          'recipe_id': '92635fd0-f2dc-4e78-a6e4-ecd556ad361f',
          'recipe_yield': '12 servings',
          'slug': 'pampered-chef-double-chocolate-mocha-trifle',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 22),
        'mealplan_id': 217,
        'recipe': dict({
          'description': 'Cheeseburger Sliders are juicy, cheesy and beefy - everything we love about classic burgers! These sliders are quick and easy plus they are make-ahead and reheat really well.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'beGq',
          'name': 'Cheeseburger Sliders (Easy, 30-min Recipe)',
          'original_url': 'https://natashaskitchen.com/cheeseburger-sliders/',
          'recipe_id': '8bdd3656-5e7e-45d3-a3c4-557390846a22',
          'recipe_yield': '24 servings',
          'slug': 'cheeseburger-sliders-easy-30-min-recipe',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.LUNCH: 'lunch'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 22),
        'mealplan_id': 216,
        'recipe': dict({
          'description': 'This All-American beef stew recipe includes tender beef coated in a rich, intense sauce and vegetables that bring complementary texture and flavor.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': '356X',
          'name': 'All-American Beef Stew Recipe',
          'original_url': 'https://www.seriouseats.com/all-american-beef-stew-recipe',
          'recipe_id': '48f39d27-4b8e-4c14-bf36-4e1e6497e75e',
          'recipe_yield': '6 servings',
          'slug': 'all-american-beef-stew-recipe',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 212,
        'recipe': dict({
          'description': 'This All-American beef stew recipe includes tender beef coated in a rich, intense sauce and vegetables that bring complementary texture and flavor.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': '356X',
          'name': 'All-American Beef Stew Recipe',
          'original_url': 'https://www.seriouseats.com/all-american-beef-stew-recipe',
          'recipe_id': '48f39d27-4b8e-4c14-bf36-4e1e6497e75e',
          'recipe_yield': '6 servings',
          'slug': 'all-american-beef-stew-recipe',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 22),
        'mealplan_id': 211,
        'recipe': dict({
          'description': 'Einfacher Nudelauflauf mit Brokkoli, Sahnesauce und extra Käse. Dieses vegetarische 5 Zutaten Rezept ist super schnell gemacht und SO gut!',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'nOPT',
          'name': 'Einfacher Nudelauflauf mit Brokkoli',
          'original_url': 'https://kochkarussell.com/einfacher-nudelauflauf-brokkoli/',
          'recipe_id': '9d553779-607e-471b-acf3-84e6be27b159',
          'recipe_yield': '4 servings',
          'slug': 'einfacher-nudelauflauf-mit-brokkoli',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 23),
        'mealplan_id': 196,
        'recipe': dict({
          'description': 'Simple to prepare and ready in 25 minutes, this vegetarian miso noodle recipe can be eaten on its own or served as a side.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': '5G1v',
          'name': 'Miso Udon Noodles with Spinach and Tofu',
          'original_url': 'https://www.allrecipes.com/recipe/284039/miso-udon-noodles-with-spinach-and-tofu/',
          'recipe_id': '25b814f2-d9bf-4df0-b40d-d2f2457b4317',
          'recipe_yield': '2 servings',
          'slug': 'miso-udon-noodles-with-spinach-and-tofu',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': FakeDate(2024, 1, 22),
        'mealplan_id': 195,
        'recipe': dict({
          'description': 'Avis aux nostalgiques des années 1980, la mousse de saumon est de retour dans une présentation adaptée au goût du jour. On utilise une technique sans faille : un saumon frais cuit au micro-ondes et mélangé au robot avec du fromage à la crème et de la crème sure. On obtient ainsi une texture onctueuse à tartiner, qui n’a rien à envier aux préparations gélatineuses d’antan !',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'rrNL',
          'name': 'Mousse de saumon',
          'original_url': 'https://www.ricardocuisine.com/recettes/8919-mousse-de-saumon',
          'recipe_id': '55c88810-4cf1-4d86-ae50-63b15fd173fb',
          'recipe_yield': '12 servings',
          'slug': 'mousse-de-saumon',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': 'Dineren met de boys',
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '3931df86-0679-4579-8c63-4bedc9ca9a85',
        'mealplan_date': FakeDate(2024, 1, 21),
        'mealplan_id': 1,
        'recipe': None,
        'title': 'Aquavite',
        'user_id': '6caa6e4d-521f-4ef4-9ed7-388bdd63f47d',
      }),
    ]),
  })
# ---
